R/tmb-methods.R

```diff
@@ -207,3 +207,111 @@ print.mmrm_tmb <- function(x,
   cat("\n")
   invisible(x)
 }
+
+
+#' Residuals method for mmrm objects
+#'
+#' @describeIn mmrm_tmb_method to obtain residuals - either unscaled or normalized
+#' @param object (`mmrm_tmb`)\cr the fitted MMRM.
+#' @param type (`character`)\cr unscaled ('response'), 'pearson' or 'normalized'. Default is 'response',
+#' and this is the only type available for use with models with a spatial covariance structure.
+#' @param ... \cr not used in this method.
+#' @return Vector of residuals
+#' @importFrom stats residuals
+#' @exportS3Method
+#' @examples
+#' # residuals:
+#' residuals(object, type = "response")
+#' residuals(object, type = "pearson")
+#' residuals(object, type = "normalized")
+#' @references
+#' - \insertRef{galecki2013linear}{mmrm}
+residuals.mmrm_tmb <- function(object, type = c("response", "pearson", "normalized"), ...) {
+  assert_class(object, "mmrm_tmb")
+  assert_character(type)
+  type <- match.arg(type, several.ok = FALSE)
+  resids_unscaled <- component(object, "y_vector") - fitted(object)
+  if (type == "response") {
+    return(unname(resids_unscaled))
```

Comment:
```suggestion
    unname(resids_unscaled)
```
do we need the `unname` here? or would it be better to do that already when defining `resids_unscaled`?